### ISSUE TITLE
Crossroads Map Graphical Issues and Lag

### ISSUE THREAD

**Issue by XARAMBIT** [PCSX2/pcsx2#3358]
_**PCSX2 version:**_ v1.5.0-dev3403-g58f5a5b91

- **Notes**:
  - There is an active community (up to 40 players sometimes) still playing SOCOM II & SOCOM CA over LAN using a TAP VPN Tunnel. Some of them are on original hardware and some are on PCSX2
  - Here is a guide outlining how everyone gets setup: https://1upsdevelopment.com/pcsx2-socom-2-xlink/
  - The games run well with these custom patches (unrelated to issue): https://harry62.1upsdevelopment.com/
  - Everyone is reporting issues with this particular map named 'CROSSROADS'

***PCSX2 options:***
``Safe Preset / Balanced``

***Plugins used:***
``GSdx 20200414184025 (MSVC 19.21 SSE4.1/AVX) 1.2.0 GSdx32-SSE4``
``LilyPad 0.12.1``
``SPU2-X 20200414184025 2.0.0``
``Qemu USB Driver 0.8.1``
``CLR DEV9 0.8.6``

- **Note**: 
Issue persists hardware or software render using default native settings or higher res

***Description of the issue:***
**The multiplayer map "CROSSROADS" has a heavy bright bloomy fog that limits visibility, but disappears upon death when spectating to make everything appear correctly again.
There is also general lag and bad performance on this map for all players on emulation
Switching to software renderer (F9) does not help**

![image](https://user-images.githubusercontent.com/52842013/80990701-c5564480-8e04-11ea-8fa2-a1f9641adda1.png)

![image](https://user-images.githubusercontent.com/52842013/80990722-cf784300-8e04-11ea-8e6f-032a1ce7fdaa.png)

***How to reproduce the issue:***
  - Reproducing this issue requires starting or being in an online game over LAN.
  - If you are testing by yourself with 1 player you will need to run PCSX2dis and execute a force code to start the LAN game with only one player or possibly convert this gameshark code to pnach (not tested pnach method) here is the ready force code: 203dd6a8 0027d070

  - Start SOCOM II
  - Open LAN if not using harry62’s r0005 patch, Login
  - CREATE GAME
  - Give the lobby any name
  - CHOOSE GAMES
  - Select Crossroads Map, and accept Play List
  - CREATE GAME
  - Wait for the ready button to be available (30 Seconds)
  - In order to run a LAN game with 1 player for testing, you need to run a ready force code
  - 203dd6a8 0027d070
  - In this case you may need to run this on PCSX2dis, I believe it is a gameshark code if you want to convert to pnach but I've not tested that method. The issues are the same whether in an actual lobby with players on PCSX2 1.5.0 or on PCSX2dis solo for testing
  - After running the force code, when you hit ready the game will begin the 10 second starting countdown timer and the map will load.

  - Alternatively you can follow 1ups guide: https://1upsdevelopment.com/pcsx2-socom-2-xlink/ to play with real players and join or host a lobby running the map CROSSROADS.

  - If you need time on the map to debug (because round timer ends within 15 seconds if only 1 player), use harry62's patch .elf to give you the ability to run the map CROSSROADS as a respawn game mode, start a solo LAN, turn on respawn*, and run the force code. The announcer will declare a team won but the timer will continue for 20 minutes allowing you to run around.

***PC specifications:***
These problems are being reported by everyone using emulation in the SOCOM Virtual Dedicated LAN (SVDL) launcher lobby chat, and is often a topic of discussion whenever this map is in rotation.

Alright thank you

**Comment by prafullpcsx2** [comment]
So the lag happens on this map only in LAN mode or lag is there when playing this map offline? Or I assume this map is not available offline since you provided ll the steps to go online?

**Comment by XARAMBIT** [comment]
Yes, this is hard for me to test since the map is only available in multiplayer (through the game menu), but if anyone knows of a way to boot the map in offline/debug mode.. I’m willing to bet the issue would persists since the lag appears to be graphical in nature. Most of the other multiplayer maps run fine with minor or no issues.

**Comment by XARAMBIT** [comment]
Yes, this is hard for me to test since the map is only available in multiplayer (through the game menu), but if anyone knows of a way to boot the map in offline/debug mode.. I’m willing to bet the issue would persists since the lag appears to be graphical in nature. Most of the other multiplayer maps run fine with minor or no issues. PCSX2 has really come a long way this is amazing

**Comment by prafullpcsx2** [comment]
I tested the game, it doesn't lag for me. This map is slightly slower than other maps. Bright bloom effect is caused in Direct 3d mode. Opengl works fine. 

![image](https://user-images.githubusercontent.com/12788199/81007543-0be08900-8e6f-11ea-9532-41a85d168cca.png)

**Comment by XARAMBIT** [comment]
Thanks for taking a look, we'll take another look at OpenGL and see if it resolves the slowdowns as well. In multiplayer, nade spam at the start of a round contributes to drastic reductions in frame rates on all maps, but that is native to the game itself. Especially with 16 players

**Comment by prafullpcsx2** [comment]
If using d3d you can use disable safe features option from gsdx hacks section to get rid of this bloom.

**Comment by xCENTx** [comment]
This issue persists through many of the maps on SOCOM 2. 
There is a patch that can be applied for the game that will fix the issue, however you will then be stuck with minimal brightness levels. (Disable Auto Eye Adjustment)

R0004 Patches
//Render Fix 1 (Created by Harry62 on r0001 , ported to r0004 by NightFyre)
//This will disable auto eye adjustment and lower the overall brightness of the game (it will be very hard to see in dark places)
2035A2F8 100000DB (original = 106000DB)

//Render Fix 2 (Created by NightFyre)
//This will fix the slowdowns completely , but the overall brightness will stay unaffected. Player aye adjustment is also retained
2035A320 00000000 (original = 0C0D64C8)

The following can be used to adjust the brightness levels of the world / environment. They take float values and the default is 0
The problem with applying a value to these addresses is that they will restore to default value if one of the following conditions are met:
- pause the game
- die
- interact with map

//Map Brightness 1
204B858C 

//Map Brightness 2
204B859C

//Map Brightness 3
204B85AC



R0001 notes
//Render Fix
2033CD68 100000DB 	[106000DB]
2033CD90 00000000 	[0C0CEF64]

//Map lighting modifier
204B4D4C 40A00000
204B4D5C 40A00000
204B4D6C 40A00000
204B4D7C 40A00000
